### ISSUE TITLE
Properly support function types

### ISSUE THREAD

**Issue by nashiora** [nashiora/laye#31]
e.g. `void(i32, f32)`. Should probably include function pointers, if there's anything that might complicate their syntax or semantics: `void(i32, f32)*` should just parse, right? and sema should be able to call both and understand that function values don't make sense most places.

**Comment by nashiora** [comment]
as a separate note, I know I wanted raw function types to be useful as generic wrappers around both static functions and closures, so `void(i32, f32)` might actually be a fat pointer type if I don't think of something else, meaning `void(i32, f32)*` is a pointer to the fat-function-pointer, not a regular function pointer...

syntactic suggestions welcome.

**Comment by nashiora** [comment]
my first thought is just to add a keyword, like `func void(i32, f32)`, giving `func void(i32, f32)*` as the function pointer variant (if the distinction is useful at that point; it is for C, but probably isn't for Laye, so it's entirely possible that `func` implies the pointer)

**Comment by MrMugame** [comment]
I don't see a reason for a new keyword. Shouldn't it just parse without like c?

**Comment by nashiora** [comment]
Correct, which is still the intent. The question is including the difference between an all-encompassing "this behaves like a function with this signature" type and a literal "this is just the type of a function (or a pointer to it)".

basically, what is the type of each of these `var`s:

```
void foo(int a);

var x = foo;
var y = (int a) => x(a);
```
